### PR TITLE
Reimplement actions skeleton

### PR DESCRIPTION
**Pull request by paulocsanz**
Actions don't run yet, but they are enqueued/dequeued, batches are created and they get processed on apply.

It doesn't have tests yet, that's also a next step in a subsequent PR.

Also fixes a bunch of misc bugs that we found along the way, that's why it's being merged even if not feature complete.

Co-Authored-By: nick@systeminit.com

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,6 +17,7 @@ deploy/docker-compose.env.yml
 app/web/cypress/screenshots/
 app/web/cypress/videos/
 bin/si/src/version.txt
+prelude
```
Comment (nickgerace) [inline .gitignore:20]: We may update the prelude, so we don't want to ignore it.

**Comment by nickgerace** [comment]
@paulocsanz approved but I think we need to ignore the `.gitignore` change.

File: .gitignore (hunk 1 above)
Comment (paulocsanz) [reply .gitignore:20]: It doesnt ignore if its already pushed

Comment (nickgerace) [reply .gitignore:20]: Nevermind. It already exists in the repo, so this is fine.

Comment (nickgerace) [reply .gitignore:20]: We commented at the same time lol

**Comment by paulocsanz** [comment]
bors merge

**Pull request merged**